dispatches/models/fossil_case/thermal_oil/thermal_oil.py

```diff
@@ -165,7 +166,10 @@ def initialize(self, state_args={}, state_vars_fixed=False,
         else:
             sopt = optarg
 
-        opt = SolverFactory(solver)
+        if solver is None:
```

Comment:
`get_solver` does this for you, as well as set the solver options. Just do `opt = get_solver(solver, optarg)`